crates/router/src/core/webhooks.rs

```diff
@@ -634,33 +636,31 @@ pub async fn trigger_webhook_to_merchant<W: types::OutgoingWebhookType>(
         W::add_webhook_header(&mut header, signature)
     }
 
-    let request = services::RequestBuilder::new()
-        .method(services::Method::Post)
-        .url(&webhook_url)
-        .attach_default_headers()
-        .headers(header)
-        .body(Some(transformed_outgoing_webhook_string))
-        .build();
+    let mut retries = WEBHOOK_RETRIES;
+    loop {
```

Comment:
Hey @trkohler, Thanks for drafting a PR. 

Though the solution for a webhook is a bit more nuanced and we could follow a bit different approach towards solving it. Let's break it down.
- [x] Write a solution to perform retries for failed webhook.
- [x] Have a defined limit on how many retires are allowed.

🎉 you have already solved for the above mentioned problems.
Now, let's understand and address some API performance and future reliability issues.
- [ ] Having a loop clause could result in a infinite loop not now (Of course!), but in the future when someone else contributes and makes some change here
- [ ] Let's consider the scenario that the merchant API might be failing a lot, this means that we are holding the connection with the *connector* (stripe, adyen, etc) for a longer duration of time, this could result in a timeout from their side.
- [ ] Finally, we are adding extra processing load on the API which should in theory be very lightweight.

**Let's look at some ways to solve these**
1. We should remove this from the critical path if the first outbound webhook fails. (How?)
2. To tackle this we can use the scheduler service that's already present. (check `crates/router/src/scheduler/workflows.rs`)
3. While having retry count and retry delay as constants, it would provide more control if we can have these as some kind of configs. (check `crates/router/src/db/configs.rs`)

> Overall the goal is to move intensive and time consuming stuff out of the critical path, and execute it somewhere else, (in case of initial failures)